release/pkg/assets/assets.go

```diff
@@ -0,0 +1,329 @@
+package assets
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	assetconfig "github.com/aws/eks-anywhere/release/pkg/assets/config"
+	assettypes "github.com/aws/eks-anywhere/release/pkg/assets/types"
+	"github.com/aws/eks-anywhere/release/pkg/filereader"
+	"github.com/aws/eks-anywhere/release/pkg/git"
+	"github.com/aws/eks-anywhere/release/pkg/images"
+	releasetypes "github.com/aws/eks-anywhere/release/pkg/types"
+	"github.com/aws/eks-anywhere/release/pkg/utils"
+)
+
+func getAssetsFromConfig(ac *assettypes.AssetConfig, rc *releasetypes.ReleaseConfig, eksDReleaseChannel, eksDReleaseNumber, kubeVersion string) ([]releasetypes.Artifact, error) {
+	var gitTag string
+	var err error
+	var artifacts []releasetypes.Artifact
+	var imageTagOverrides []releasetypes.ImageTagOverride
+	projectName := ac.ProjectName
+	projectPath := ac.ProjectPath
+	sourcedFromBranch := rc.BuildRepoBranchName
+	gitTagPath := projectPath
+	if ac.HasSeparateTagPerReleaseBranch {
+		gitTagPath = filepath.Join(projectPath, eksDReleaseChannel)
+	}
+	// Get git tag for project if exists
+	if !ac.NoGitTag {
+		if projectName == "eks-anywhere-cluster-controller" {
```

Comment:
This looks like an anti-pattern to me
We are trying to dry out common logic, moving from individual implementations to a common shared implementation, from concrete to abstract. But this is including the concrete (a particular project name) in the abstract (the shared "oracle"